util/PostgresMigrations/HelperScripts/2024-08-21_00_FinalFlexibleCollectionsDataMigrations.psql

```diff
@@ -0,0 +1,32 @@
+﻿-- Migrate Custom users who only have 'editAssignedCollections' and/or 'deleteAssignedCollections' custom permissions to the User type.
+UPDATE "OrganizationUser"
+SET
+    "Type" = 2,
+    "Permissions" = NULL
+WHERE
+    "Type" = 4
+    AND jsonb_typeof("Permissions"::jsonb) = 'object'
+    AND (
+        ("Permissions"::jsonb)->>'editAssignedCollections' = 'true'
+        OR ("Permissions"::jsonb)->>'deleteAssignedCollections' = 'true'
+    )
+    AND COALESCE(("Permissions"::jsonb)->>'accessEventLogs', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'accessImportExport', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'accessReports', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'createNewCollections', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'editAnyCollection', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'deleteAnyCollection', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'manageGroups', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'managePolicies', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'manageSso', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'manageUsers', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'manageResetPassword', 'true') = 'false'
+    AND COALESCE(("Permissions"::jsonb)->>'manageScim', 'true') = 'false';
```

Comment:
You removed some but not all of the `COALESCE`s in this file - I'm thinking these are also unnecessary now.